core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
I think it was, or at least, that's what we were discussing at https://github.com/apache/spark/pull/11242#discussion_r56891305 Not sure what the original in this PR was because it was force-pushed. In any event, I personally agree that a minimum count has a little more motivation, even if I still don't see when this setting would be used. Where do we disagree ... are you saying that a problematic `InputFormat` of this form would actually work everywhere else in Spark?